### PR TITLE
chore: updated s390x builder

### PR DESCRIPTION
**Pull request by ksdeekshith**
# Description
Updated builder for IBM/Z 

## Tests that have been done

-  Verified the Building and publishing docker build through [workflow](https://github.com/ksdeekshith/quay/actions/runs/5174142121/jobs/9320128036) 

## Changes that have been done

-  Updated **set-output** command, which is being deprecated, to **GITHUB_OUTPUT**
-  Removed **QEMU**
-  Added on-the-fly **creation of ZVSI for s390x builder** and **deletion post build-and-publish-s390x**

### CODE REVIEW

File: .github/workflows/build-and-publish.yaml [hunk 1]
```diff
@@ -29,11 +39,56 @@ jobs:
...
+              apt-get install ca-certificates curl gnupg
+              install -m 0755 -d /etc/apt/keyrings
+              curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+              chmod a+r /etc/apt/keyrings/docker.gpg
+              echo \
+              "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+              "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+              tee /etc/apt/sources.list.d/docker.list > /dev/null
+              apt-get update
+              apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin -y
```
Comment (yussufsh) [inline .github/workflows/build-and-publish.yaml:90]: Creating a custom image/snapshot where this is already set up will help reduce the build time.

File: .github/workflows/build-and-publish.yaml [hunk 2]
```diff
@@ -99,4 +169,12 @@ jobs:
           platforms: linux/amd64,linux/arm64,linux/ppc64le,linux/s390x
           push: true
           tags: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}:${{ github.event.inputs.tag || env.TAG }}
-
+      
+      - name: Post Build-and-publish-s390x
+        run: |
+          #Delete the created ZVSI
+          ibmcloud is instance-delete $ZVSI_INSTANCE_NAME --force
+          sleep 15
```
[the remark this answers is not in the document]
Comment (ksdeekshith) [reply .github/workflows/build-and-publish.yaml:177]: The instance will take time to be deleted. And updated to 20 sec which will be enough.

File: .github/workflows/build-and-publish.yaml [hunk 3]
```diff
@@ -99,4 +169,12 @@ jobs:
           platforms: linux/amd64,linux/arm64,linux/ppc64le,linux/s390x
           push: true
           tags: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}:${{ github.event.inputs.tag || env.TAG }}
-
+      
+      - name: Post Build-and-publish-s390x
+        run: |
```
[the remark this answers is not in the document]
Comment (ksdeekshith) [reply .github/workflows/build-and-publish.yaml:174]: will update the code with the following check
`if: ${{ steps.ZVSI.conclusion == 'success' && always() }}`

File: .github/workflows/build-and-publish.yaml [hunk 4]
```diff
@@ -29,11 +39,56 @@ jobs:
...
+          #Saving the Floating IP to login ZVSI
+          ZVSI_HOST=$(ibmcloud is floating-ip $ZVSI_FP_NAME | awk '/Address/{print $2}')
+          echo $ZVSI_HOST
+          echo "IP=${ZVSI_HOST}" >> $GITHUB_OUTPUT
+          
+      - name: Setup Docker on ZVSI
+        uses: appleboy/ssh-action@master
+        with:
+            host: ${{ steps.ZVSI.outputs.IP }}
+            username: ${{ secrets.SSH_USER }}
```
[the remark this answers is not in the document]
Comment (ksdeekshith) [reply .github/workflows/build-and-publish.yaml:73]: Sure will update with **ZVSI_SSH_USER**

File: .github/workflows/build-and-publish.yaml (hunk 1 above)
Comment (ksdeekshith) [reply .github/workflows/build-and-publish.yaml:90]: yeah, that will be a good option that reduces the build time. will update the custom image which is pre-installed with docker.

Thanks for the tip ✌️

**Review by dmage — APPROVED** [review]

**Pull request merged**